_posts/2016-10-06-Introduction-to-Share-Action-Extensions.md

```diff
@@ -0,0 +1,69 @@
+
+
+There are two types of extensions present in iOS apps: share and action extensions. Both are valuable in your daily workflows and developers created unique ways to interact with them. Use share extensions are to share text, images or videos to social networks or other apps. Think of it as *leaving the current app and continuing in another*. Use action extensions to *modify content in place without leaving the app*. Examples of share extensions are:
+
+- Share to (Facebook, Twitter, Reminders, Notes etc.)
+- Import with (Dropbox, Mailchimp, PDF Expert etc.)
+
+![](/img/2016-10-11-Share-Action-Extensions-0.PNG)
+
+While this looks straight forward everything that does not fit into the above list should be an action extension:
+
+- Print
+- Copy
+- Run Workflow
+- Save Image
+
+![](/img/2016-10-11-Share-Action-Extensions-1.PNG)
+
+They can be application specific. A developer can decide if his action extension is only available in their app or in any other. This is in the hands of developers. The only way to get it out of their apps is to talk to them and ask them to make it available.
+
+## Share Extensions
+
+Any action in a share extension will take the selected content and send them to another app by either leaving the current app or by displaying a share sheet:
+
+![](/img/2016-10-11-Share-Action-Extensions-2.jpg)
+
+Developers can provide custom UI, like Notes, or use the standard look and feel, like Twitter.
+
+There is a third kind of share extensions that gets automatically added when a developer supports certain types of files. These follow the naming scheme *Copy to…* (before iOS 10) and *Import with…* (iOS 10). These take the selected file and copy them into the selected app. This is similar to document providers discussed in our [Introduction to Document Providers](http://mobileonly.tools/blog/2016/10/06/Introduction-to-Document-Providers.html) article.
+
+![](/img/2016-10-11-Share-Action-Extensions-3.jpg)
+
+## Action Extensions
+
+Share extensions follow some kind of order but with action extensions, it feels like the wild west. Developers can add application specific action extensions, some action extensions should have been share extensions, some leave the app, some stay within the app. But with great chaos comes great possibilities.
+
+Great examples of application wide action extensions are *Print* by Apple and *Run Workflow* by [Workflow](https://itunes.apple.com/us/app/workflow-powerful-automation/id915249334?mt=8), let us see what they can do for us.
+
+### Print
+
+Apple introduced AirPrint in iOS 4.2 along with the iPad. Manufacturers released a few printers but they are not taking them very seriously. Users on the other side do not care too much either. So why talk about it at all? People coming from macOS will miss a function called "Print as PDF" known from the print dialogue. Until iOS 10 developers had to build it themselves, but now Apple integrated it into the standard AirPrint action extension.
```

Comment:
A bit unclear who ist taking what not seriously. Maybe:
"Manufacturers released a few COMPATIBLE(?) printers but they are not taking them very seriously."